refactor: disclosure list item icon placement to right

kendallstrautman opened the pull request:
## 🔗 Relevant links

<!--
Include links to the branch preview, Asana task, and Figma designs wherever possible to make reviewing your PR easier.
When including the preview link, make sure to remove any '.' characters from the branch name:
  - Example: ks.my-branch -> ksmy-branch
-->

- [Preview link](https://dev-portal-git-ksrefactor-mobile-menu-icon-placement-hashicorp.vercel.app/boundary) 🔎
- [Asana task](https://app.asana.com/0/1203652232454021/1204303889362367) 🎟️
- [Figma](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?node-id=815-40857&t=BLzfEIgRtHWvgFn3-0) 🎨 

## 🗒️ What

This PR refactors the placement of the dropdown disclosure item icon to the right hand side. 

<img width="326" alt="Screenshot 2023-04-03 at 5 58 52 PM" src="https://user-images.githubusercontent.com/36613477/229659160-ac64272d-c58d-47ae-aa9e-aff7086fde48.png">
<img width="478" alt="Screenshot 2023-04-03 at 5 59 04 PM" src="https://user-images.githubusercontent.com/36613477/229659180-2ce98274-09e5-452d-9fe3-1318ec5c95d8.png">

## 🧪 Testing

- [Visit the preview](https://dev-portal-git-ksrefactor-mobile-menu-icon-placement-hashicorp.vercel.app/boundary)
- Sign in
- Test the desktop dropdown disclosure placement
- Test the tablet / mobile view

## 💭 Anything else?

We've removed all the icons from the menus except the external link icon. [This is a follow up task.](https://github.com/hashicorp/dev-portal/pull/1793#issuecomment-1491047999)

kendallstrautman commented on src/components/dropdown-disclosure/components/list-item/index.tsx:
```diff
@@ -97,10 +97,10 @@ const DropdownDisclosureButtonItem = ({
 				<Text asElement="span" size={200} weight="medium">
 					{children}
 				</Text>
+				{icon}
```
Need to validate whether I can globally change this icon placement. I _think_ its okay.

kendallstrautman replied:
Ah yes, these two places are where we want the icon to flip so thats good!

zchsh reviewed (approved):
✅ Followed the testing steps, seeing everything as described 👍

braydenlove reviewed (approved):
LGTM

braydenlove reviewed (approved):
Looks good! 🚢 it!

kendallstrautman commented:
@ashleemboyer I agree I think that could eventually be a nice thing to add. Since we only need it on one side for this work, I'd prefer that to be a follow up task given my current timeframe if possible.

kendallstrautman commented:
@ashleemboyer i'm going to merge this in but I added a[ follow up ticket](https://app.asana.com/0/1203652232454021/1204332555573341) to capture the icon position idea and added to our eng backlog 😄
